### PR TITLE
351 water pipeline repair cost analysis

### PR DESCRIPTION
**Pull request by longshuicy**
Pipeline repair cost is different than EPF and Water Facility as it's chaining with the pipeline damage rate to determine number of leaks and number of breaks. 

Then further reading the damage ratio table to calculate the aggregated repair cost. 

Need to verify the number with Hesam. The repair cost seems to be small comparing to facility and epf. 

To test: run the `tests/pyincore/analyses/pipelinerepaircost/test_pipelinerepaircost.py`

### CODE REVIEW

**Comment by ywkim312** [comment]
Is it okay to just run the test? What about the values comparison part?

File: pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+import concurrent.futures
+
+from pyincore import AnalysisUtil
+from pyincore import BaseAnalysis
+
+
+class PipelineRepairCost(BaseAnalysis):
+    """Computes electric power facility repair cost.
```
Comment (navarroc) [inline pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py:14]: pipeline

File: pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                {
+                    "id": "diameter",
+                    "required": False,
+                    "description": "pipeline diameter cutoff for different damage ratio",
+                    "type": int
+                },
+                {
+                    "id": "segment_length",
+                    "required": False,
+                    "description": " 20-foot segments in the pipeline",
```
Comment (navarroc) [inline pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py:182]: Maybe change this description to "Segment length assumption. Default is 20 feet".

File: pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                {
+                    "id": "num_cpu",
+                    "required": False,
+                    "description": "If using parallel execution, the number of cpus to request.",
+                    "type": int
+                },
+                {
+                    "id": "diameter",
+                    "required": False,
+                    "description": "pipeline diameter cutoff for different damage ratio",
```
Comment (navarroc) [inline pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py:176]: Maybe change this to "Pipeline diameter cutoff assumption for different damage ratios. Default is 20 inches"

File: pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+            segment_length = 20  # 20 feet
+
+        diameter = self.get_parameter("diameter")
+        if diameter is None:
+            diameter = 20  # 20 inch
+
+        repair_costs = []
+
+        for pipeline in pipelines:
+            pipe_length = pipeline["length"]  # kilometre
```
Comment (navarroc) [inline pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py:101]: kilometer?

**Comment by longshuicy** [comment]
> test generated three files with values. seaside_eq_pipeline_result.csv, seside_eq_pipeline_result_additional_info.json, and seaside_pipeline_repair_cost.csv.
> 
> @longshuicy If the datatype repaircost is a new datatype, can you create an issue or PR to include this in incore-data-release?
> 
> approve

https://github.com/IN-CORE/incore-data-release/pull/53 here is the PR

File: pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py (hunk 4 above)
Comment (longshuicy) [reply pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py:101]: hahaha maybe I am British!
jkjk will correct the typo

**Review by navarroc — APPROVED** [review]

**Pull request merged**